### PR TITLE
Tests for data export form

### PR DESCRIPTION
**Pull request by agrawalarpit14**
Add tests, correct minor errors and remove dependency of data export on cookies.
Part Of #582

### CODE REVIEW

**Comment by agrawalarpit14** [comment]
Hi @bkimminich
This PR contains tests for the data export component. Presently api tests for the requests containing previous orders are not there.
For the tests with orders shall I consider the following flow.
- User logs in.
- User adds product to the basket.
- User checks-out to complete the purchase.
- User request data export.

An example of above description is as follows
```
return frisby.timeout(10000).post(REST_URL + '/user/login', {
      headers: jsonHeader,
      body: {
        email: 'bjoern.kimminich@googlemail.com',
        password: '****************'
      }
    })
      .expect('status', 200)
      .then(({json: jsonLogin}) => {
        return frisby.post(API_URL + '/BasketItems', {
          headers: { 'Authorization': 'Bearer ' + jsonLogin.authentication.token, 'content-type': 'application/json' },
          body: {
            BasketId: 5,
            ProductId: 2,
            quantity: 1
          }
        })
          .expect('status', 200)
          .then(() => {
              return frisby.post(REST_URL + '/basket/5/checkout', {
                headers: { 'Authorization': 'Bearer ' + jsonLogin.authentication.token, 'content-type': 'application/json' }
              })
                .expect('status', 200)
                .then(() => {
                    return frisby.post(REST_URL + '/data-export', {
                      headers: { 'Authorization': 'Bearer ' + jsonLogin.authentication.token, 'content-type': 'application/json' },
                      body: {
                        format: "1"
                      }
                    })
                      .expect('status', 200)
                      .expect('header', 'content-type', /application\/json/)
                      .expect('json', 'confirmation', 'Your data export will open in a new Browser window.')
                      .expect('json', 'userData', '{\n  \"username\": \"\",\n  \"email\": \"bjoern.kimminich@googlemail.com\"\n}')
                  })
            })
      })
```
I had following questions for the addition of these tests.
- Can the flow of the tests be reduced(in terms of number of api calls)?
- Is the way in which the comparison of response is done acceptable(the last line of above code snippet)?

Please let me know of your views and suggestions.
Thank You!!

**Comment by bkimminich** [comment]
Comparing the relevant pieces of the response is the right way to check, yes. Also for complex flows you must either rely on the data which is in the DB my default (from datacreator) or create it yourself in this kind of bloated API call sequence. There's no other way as the tests cannot depend on each other due to their async nature. Also that'd be bad testing practice even if it worked.

**Comment by agrawalarpit14** [comment]
> Comparing the relevant pieces of the response is the right way to check, yes. Also for complex flows you must either rely on the data which is in the DB my default (from datacreator) or create it yourself in this kind of bloated API call sequence. There's no other way as the tests cannot depend on each other due to their async nature. Also that'd be bad testing practice even if it worked.

Sure, then I will go on and write tests considering orders as well.
Thank You!!

**Comment by agrawalarpit14** [comment]
Hi @bkimminich I have added the tests corresponding to orders as well.
Please let me know your feedback.
Thank You!!😁😆

File: test/api/dataExportApiSpec.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+              body: {
+                format: '1'
+              }
+            })
+              .expect('status', 200)
+              .expect('header', 'content-type', /application\/json/)
+              .expect('json', 'confirmation', 'Your data export will open in a new Browser window.')
+              .then(({ json }) => {
+                // eslint-disable-next-line
+                expect(json.userData.replace(/(\n)/gm, '')).toMatch(/\{ {2}\"username\": \"\", {2}\"email\": \"jim@(.*)\", {2}\"orders\": \[ {4}\{ {6}\"orderId\": \"(.*)\", {6}\"totalPrice\": 9.98, {6}\"products\": \[ {8}\{ {10}\"quantity\": 2, {10}\"name\": \"Raspberry Juice \(1000ml\)\", {10}\"price\": 4.99, {10}\"total\": 9.98, {10}\"bonus\": 0 {8}\} {6}\], {6}\"bonus\": 0, {6}\"eta\": \"(.*)\" {4}\} {2}\]\}/)
```
Comment (J12934) [inline test/api/dataExportApiSpec.js]: Could we change this to parse the json and then run the assertions against the parsed object?

Regexes are giving me nightmares 😉

Comment (bkimminich) [reply test/api/dataExportApiSpec.js]: I couldn't agree more... :laughing:

**Comment by agrawalarpit14** [comment]
Hi @bkimminich @J12934 I have removed the `RegExes` and parsed the `JSON` to validate the tests.
Actually it was much better to parse.(but still I got to learn a lot about RegExes)😅
Please let me know your feedback.
Thank You!!😁😆😎

**Pull request merged**